stencil-push.sh

```diff
@@ -0,0 +1,58 @@
+set -e
+
+project_version=$1
+proto_repo=$2
+file_path=$3
+output_socket=/dev/null
+
+log_info()  { echo "$(date) - [INFO] $@"; }
+log_warn()  { echo "$(date) - [WARN] $@"; }
+log_error() { echo "$(date) - [ERROR] $@"; }
+
+help() {
+    echo ""
+    echo "stencil-push.sh <version> <proto-repo> <file-path>"
+    echo "   version      - version of the proto descriptor set being uploaded"
+    echo "   proto-repo   - repository name where to upload the proto descriptor set"
+    echo "   file-path    - path to the proto descriptor set binary file"
+    echo "   "
+    echo "   stencil-push.sh needs a few ENV variables set - STENCIL_HOSTNAME, STENCIL_USERNAME, STENCIL_PASSWORD"
+    echo "   STENCIL_HOSTNAME - Hostname serving the stencil APIs"
+    echo "   STENCIL_USERNAME - HTTP Basic Auth username"
+    echo "   STENCIL_PASSWORD - HTTP Basic Auth password"
+}
+
+if [[ -z "$STENCIL_HOSTNAME" ]] || [[ -z "$STENCIL_USERNAME" ]] || [[ -z "$STENCIL_PASSWORD" ]]
+then
+    log_error "Missing values for ENV variables"
+    help
+    exit 1
+fi
+if [ $# -lt 3 ] || [ $# -gt 3 ]
+then
+    if [ $# -eq 1 ] && [ $1 == "help" ]
+    then
+        help
+        exit 0
+    fi
+    log_error "Command requires exactly 3 arguments, $# were passed"
+    log_error "Passed arguments - " $@
+    help
+    exit 1
+fi
+
+trap 'log_error "Failed: $ACTION"' EXIT
+
+ACTION="Uploading proto descriptor as version $project_version"
+log_info $ACTION
+curl -sS -w "\n" -u $STENCIL_USERNAME:$STENCIL_PASSWORD -X PUT --fail "https://$STENCIL_HOSTNAME/artifactory/proto-descriptors/$proto_repo/$project_version" -T "$file_path" > /dev/null
```

Comment:
The URL could be parametrized as the users decide the location of the artifact?
If you agree , the other scripts could be updated and the comments accordingly.

Reply:
Could be done when a users requires it.